srv/modules/runners/validate.py

```diff
@@ -278,37 +278,41 @@ def storage(self):
 
     def ganesha(self):
         """
-        Nodes may only be assigned one ganesha role.  Ganesha depends on 
+        Nodes may only be assigned one ganesha role.  Ganesha depends on
         cephfs or radosgw.
         """
         ganesha_roles = []
         role_mds = False
         role_rgw = False
+        role_ganesha = False
 
-        for node in self.data.keys():
-            if ('roles' in self.data[node]):
-                if('ganesha_configurations' in self.data[node]):
-                    ganesha_roles = list(set(self.data[node].get("roles")) &
-                                        set(self.data[node].get("ganesha_configurations")))
+        for node, data in self.data.items():
+            if ('roles' in data):
+                if('ganesha_configurations' in data):
+                    ganesha_roles = list(set(data.get("roles")) &
+                                        set(data.get("ganesha_configurations")))
                     if len(ganesha_roles) > 1:
                         msg = "minion {} has {} roles. Only one permitted".format(node, ganesha_roles)
                         self.errors.setdefault('ganesha',[]).append(msg)
 
 
                 if not (role_mds or role_rgw):
-                    if('mds' in self.data[node]['roles']):
+                    if('mds' in data['roles']):
                         role_mds = True
-                    if('rgw' in self.data[node]['roles']):
+                    if('rgw' in data['roles']):
                         role_rgw=True
-                    if('rgw_configurations' in self.data[node]):
-                        if(list(set(self.data[node].get("roles")) &
-                                set(self.data[node].get("rgw_configurations")))):
+                    if('rgw_configurations' in data):
+                        if(list(set(data.get("roles")) &
+                                set(data.get("rgw_configurations")))):
                             role_rgw=True
 
-        if not (role_mds or role_rgw):
+                if not role_ganesha:
+                    role_ganesha = 'ganesha' in data['roles']
+
+        if not (role_mds or role_rgw) and role_ganesha:
             msg = "Ganesha requires either mds or rgw node in cluster."
             self.errors['ganesha'] = msg
```

Comment:
We can set role_ganesha True in the previous loop. 
Maybe rewrite as. 
if role_ganesha:
  if(role_rgw or role_mds):
    self.passed['ganesha'] = "valid"
  else:
     msg = "Ganesha requires either mds or rgw node in cluster."
     self.errors['ganesha'] = msg